contrib/caldav/calcurse-caldav.py

```diff
@@ -233,7 +316,10 @@ def push_object(conn, objhash):
         return None
 
     etag = None
-    headerdict = dict((key.lower(), value) for key, value in headers)
+    if authmethod == 'basic':
+        headerdict = dict((key.lower(), value) for key, value in headers)
+    elif authmethod == 'oauth2':
+        headerdict = dict((key.lower(), headers[key]) for key in headers)
```

Comment:
Same as above. Looks like `headers` has a different type here, depending on whether OAuth2 is used or not. This might be superfluous if we switch basic authentication to use httplib2 as well first. But even if we do not, we should make `headers` always be a proper dictionary when it is set instead. Maybe a simple cast is enough.

Reply:
You're right. I had a bit of a brain fart and could have just done a simple cast, as you mentioned, to convert it from an httplib2.Response object to a proper dict. I will fix this.